src/main/kotlin/com/expediagroup/sdk/core/configuration/collector/ConfigurationCollector.kt

```diff
@@ -16,15 +16,22 @@
 package com.expediagroup.sdk.core.configuration.collector
 
 import com.expediagroup.sdk.core.configuration.provider.ConfigurationProvider
+import com.expediagroup.sdk.core.constant.Configuration.AUTH_ENDPOINT
+import com.expediagroup.sdk.core.constant.Configuration.ENDPOINT
+import com.expediagroup.sdk.core.constant.Configuration.KEY
+import com.expediagroup.sdk.core.constant.Configuration.SECRET
+import com.expediagroup.sdk.core.constant.provider.LoggingMessageProvider
+import com.expediagroup.sdk.core.plugin.logging.OpenWorldLoggerFactory
 
 /**
  * Configuration collector that collects configuration from all available providers.
  *
  * @param providers A configuration providers queue.
  */
-internal class ConfigurationCollector private constructor(providers: ConfigurationProviderQueue) : ConfigurationProvider {
+internal class ConfigurationCollector private constructor(providers: ConfigurationProviderQueue) {
```

Comment:
Why do you remove the interface? We need this to implement the `ConfigurationProvider` interface. Please undo this change.

Reply:
It is a `ConfigurationProvider` that collects the configurations from other providers. Furthermore, we needed to make it implement the `ConfigurationProvider` interface to make sure it's up-to-date with any changes in the providers so that one should not need to access any `ConfigurationProvider` implementations directly (but via the `ConfigurationCollector`).

For it to have a `name`, we have 2 options:
1. Let it have a `name`, even if it's not used.
2. Revert back to the solution we discussed earlier of having an external map of the names instead of adding a `name` property to the interface.

I prefer the first option.

Reply:
Alright, that makes sense. I will be sticking with the first option and give it a name